cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -355,14 +355,27 @@ pub enum TypedManifest {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
 #[cfg_attr(test, derive(proptest_derive::Arbitrary))]
 #[serde(deny_unknown_fields)]
+// We use skip_serializing_if throughout to reduce the size of the lockfile and
+// improve backwards compatibility when we introduce fields.
+// We don't use Option and skip_serializing_none because an empty table gets
+// treated as Some,
+// but we don't care about distinguishing between a table not being present and
+// a table being present but empty.
+// In both cases, we can just skip serializing.
+// It would be better if we could deny_unknown_fields when we're deserializing
+// the user provided manifest but allow unknown fields when deserializing the
+// lockfile,
+// but that doesn't seem worth the effort at the moment.
```

Comment:
nit (nb)

```
/// <doc comments>
#[attrs]
// <more nondoc comments>
```

looks a bit unconventional, but the alternative might not be a whole lot clearer either.

```
/// <doc comments>
// <more nondoc comments>
#[attrs]
```

Reply:
Moved it above and separated with a newline